In [0]:
import gspread
import json
import pandas as pd

In [0]:
cred = {
  "type": "service_account",
  "project_id": "sinuous-transit-193207",
  "private_key_id": "9abb00607604068816840bb1f96256730777f3a0",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDoW1grRWd6PrGZ\nY97HwjwmE6u+iUbfTmyC5UO+rRcMYRe+eDmOrUvKVLy348vK9Wly/XD5W8kychtK\n2zX+MHMwhv3vo0mcu/YYWRkxmM9axTNNsOiPF9ZvMsgg0yZMUmSXUfF9FM348iNp\nWHR8qI9w2+hUQfPLBUBD4gL0uBbFx9osALnGJeN1eBcexi49lrHwZYKbGlNg1wru\no0ZvkTptXa35IMpo+qX8OSzHfGdk8f8t4XZroMVz4Bnq/Nn+IRksfBHr4k1JEauv\nKX2ClnPIHHEb1z7HP78INnX1JuXgJUQM+LghaXtohkADMb+09erM9ztCIn3N8jAj\nrguNUpWZAgMBAAECggEABOK2zeIdtOJQlasV6hlZbmuHdvAn/1YEzhPIAuBc1bPM\n7gJVI3zdVRnHBg7SIChc2UAe+f9GEbt4e3B0iC24nlKYUEucpqEFIant768I3R28\nYGvR2ABmT9w/fQEzkvNAjjX43+pqEkv1Fqf3ew+IRJU7I1pnp+MqYMMF0ON2Bx6M\nY5UbGHejFbOvC3/dlBbHOnhHQwwLRK0Z6Mk9fYvslv+oWZQ/kKC7Oumy/2YkGKvg\nekQee2fSB9diYkz3dhEdY3RlKYdeGyWHrNDW63/steI4cZawOjvcJgNBFupb2RdI\nU3RteG6mNZoafZAhhL1LC6Wb5Ygy4iIay0aeaWR26QKBgQD5JfXaw+DzCxK5yA/q\ncBfdtnMUYBl5NboRbpf55x8M6b5eAiHnBngY7XIeC+4IPNIqFCLauwAAA8buoLkz\ntq/9r3gOqXgFrdsyWltPPX6bu31ebDTWz+ssQo8uFZpZ/LTgV9+8QDf9yF7qH/vc\n/Qtb03Y8VQ3QN39sSJZxRruQzwKBgQDuvytxG8U1RpWHGrkJejOH031w9Oikh+7L\nE7VrOckYiSZOntOTVpCyR3PYp0fe0e3Us9QE7YGqYJoOXhPDcJPDUxIEY2KLgUWO\nEvHmICcJ332SiSd0zJXlR87VCCaZXddrAGICCUx+sY77SmMDn/xu2K+GVLnxJg5G\nA7NXZOT9FwKBgQDG2Uc5zPAyynmtkUEyqThDGiUK9BAgr4LL+uqM56vekIgBW85X\nJtCfhJN566eVJ4XNU7jlvi4n1NvnBJwL7BjWbj9x5EwIrFOwAnGx8rCj9qX+wHoD\noHpgHtdLvq4seAacr0XAZ7jIci1eyxwnqkZwiXtqd6/IqSKn998qlXuJ/QKBgDhx\njc2x6Q/NXjCQyspIz8H+6Bs6a33qfTQZtrxg7wAT/LB9sDuXLaJYbjT8Z2twai/t\nS+TWWd9z8zgOWdHM3cJB83KCJJwitdT++Z2UYGZfJgrszh04+P852NZYlPeRTgqI\nDqO4mxMg21LhpUMIId/67pRtl4BBpQI8gbvZqKGFAoGASjAlQX9J8/cusFZigRSi\nQ9xeoz7Axc09g5gIkn10ZtDWkkR4Y4phmvo1BWx3hIcz4yHhH4Nbxc6oxOPm6DeX\ntOnRNABbxaESAt/a86HVQP0qB0vWrqwBqXyk417/LBEblc9HRqFc1+mv9K5RQos+\nK2w71pLKBc0KiKix+xKEioM=\n-----END PRIVATE KEY-----\n",
  "client_email": "gspread@sinuous-transit-193207.iam.gserviceaccount.com",
  "client_id": "117309778584715418012",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/gspread%40sinuous-transit-193207.iam.gserviceaccount.com"
}
with open('9abb00607604.json', 'w') as f:
  json.dump(cred, f)

In [0]:
from oauth2client.service_account import ServiceAccountCredentials

scope = ['https://spreadsheets.google.com/feeds',
         'https://www.googleapis.com/auth/drive']

credentials = ServiceAccountCredentials.from_json_keyfile_name('9abb00607604.json',scope)

gc = gspread.authorize(credentials)

In [0]:
sht1 = gc.open_by_key('1J8LV_Lh1383TIhWp1aO0oZlESHqqLrUmT1csOg6yHXI')

In [0]:
worksheet = sht1.worksheet('Data1')

In [0]:
# Get the formula value from the souce cell:
cell_value = worksheet.get_all_values()
print(cell_value)

[['Timestamp', 'Email Address', 'Worker Name', 'M/C', 'Inspection date', 'Partname / Model', 'Material', 'Color name', 'Item code', 'Po. No.', 'Lot size', "Q'TY Cavity ", 'Drawing No.', 'Document. No.', 'Contents of inspection [เช็คความถูกต้องของใบสั่งกับชิ้นงาน]', 'Contents of inspection [ความเหมือนของสี]', 'Contents of inspection [การประกอบร่วม]', 'Contents of inspection [รอยประกบต้องไม่เกิน]', 'Problem', 'STD A', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD B', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD C', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD D', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD E', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD F', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD G', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD H', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD I', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD J', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD K', 'No.1', 'No.2', 'No.3', 'No.4', 'Result', 'STD L', 'No.1', 'No.2', 'No

In [0]:
df = pd.DataFrame(cell_value)
df.head()

0                   1   ...             92        93
0  Timestamp       Email Address  ...  Inspection by  Approve 
1   3/6/2020  sawai@hatari.co.th  ...      สุรีรัตน์    Accept

[2 rows x 94 columns]

In [0]:
worker_name = df.iloc[1,2]
machine_name = df.iloc[1,3]
inspection_date = df.iloc[1,4]
partname_model= df.iloc[1,5]
document_no = df.iloc[1,13]
inspection_date = df.iloc[1,4]
item_code = df.iloc[1,8]
color_name = df.iloc[1,7]
material= df.iloc[1,6]
po_number= df.iloc[1,9]
lot_size= df.iloc[1,10]
qty_cavity= df.iloc[1,11]
drawing_number= df.iloc[1,12]



problem = df.iloc[1,18]

STD_A = df.iloc[1,19]
A1 = df.iloc[1,20]
A2 = df.iloc[1,21]
A3 = df.iloc[1,22]
A4 = df.iloc[1,23]
result_A = df.iloc[1,24]

STD_B = df.iloc[1,25]
B1 = df.iloc[1,26]
B2 = df.iloc[1,27]
B3 = df.iloc[1,28]
B4 = df.iloc[1,29]
result_B = df.iloc[1,30]

STD_C = df.iloc[1,31]
C1 = df.iloc[1,32]
C2 = df.iloc[1,33]
C3 = df.iloc[1,34]
C4 = df.iloc[1,35]
result_C = df.iloc[1,36]

STD_D = df.iloc[1,37]
D1 = df.iloc[1,38]
D2 = df.iloc[1,39]
D3 = df.iloc[1,40]
D4 = df.iloc[1,41]
result_D = df.iloc[1,42]

STD_E = df.iloc[1,43]
E1 = df.iloc[1,44]
E2 = df.iloc[1,45]
E3 = df.iloc[1,46]
E4 = df.iloc[1,47]
result_E = df.iloc[1,48]

STD_F = df.iloc[1,49]
F1 = df.iloc[1,50]
F2 = df.iloc[1,51]
F3 = df.iloc[1,52]
F4 = df.iloc[1,53]
result_F = df.iloc[1,54]

STD_G = df.iloc[1,55]
G1 = df.iloc[1,56]
G2 = df.iloc[1,57]
G3 = df.iloc[1,58]
G4 = df.iloc[1,59]
result_G = df.iloc[1,60]

STD_H = df.iloc[1,61]
H1 = df.iloc[1,62]
H2 = df.iloc[1,63]
H3 = df.iloc[1,64]
H4 = df.iloc[1,65]
result_H = df.iloc[1,66]

STD_I = df.iloc[1,67]
I1 = df.iloc[1,68]
I2 = df.iloc[1,69]
I3 = df.iloc[1,70]
I4 = df.iloc[1,71]
result_I = df.iloc[1,72]

STD_J = df.iloc[1,73]
J1 = df.iloc[1,74]
J2 = df.iloc[1,75]
J3 = df.iloc[1,76]
J4 = df.iloc[1,77]
result_J = df.iloc[1,78]


STD_K = df.iloc[1,79]
K1 = df.iloc[1,80]
K2 = df.iloc[1,81]
K3 = df.iloc[1,82]
K4 = df.iloc[1,83]
result_K = df.iloc[1,84]

STD_L = df.iloc[1,85]
L1 = df.iloc[1,86]
L2 = df.iloc[1,87]
L3 = df.iloc[1,88]
L4 = df.iloc[1,89]
result_L = df.iloc[1,90]

inspection_by = df.iloc[1,91]
remark = df.iloc[1,92]
status = df.iloc[1,93]

no1 = df.iloc[1,14].split(',')
no2 = df.iloc[1,15].split(',')
no3 = df.iloc[1,16].split(',')
no4 = df.iloc[1,17].split(',')


no1[0]
no1[1]
#no1[2]
#no1[3]

no2[0]
no2[1]
#no2[2]
#no2[3]

no3[0]
no3[1]
#no3[2]
#no3[3]

no4[0]
no4[1]
#no4[2]
#no4[3]

M1 = no1[0]
M2 = no1[1]
#M3 = no1[2]
#M4 = no1[3]

N1 = no2[0]
N2 = no2[1]
#N3 = no2[2]
#N4 = no2[3]

O1 = no1[0]
O2 = no1[1]
#O3 = no1[2]
#O4 = no1[3]

P1 = no1[0]
P2 = no1[1]
#P3 = no1[2]
#P4 = no1[3]



In [0]:

output_template = '''
|WANAVIT MANUFACTURING CO.,LTD  |
|:-------------------------------:|
| **IPQC. PLASTIC DATA SHEET : ( First Trial )**  |

| Worker Name :{worker_name}  | M/C:{machine_name}     |  Partname/Model :{partname_model}   | Document .No. :{document_no} |
|----------|-------------|------|------|
| **Inspection date&Time :{inspection_date}**|  **Item Code :{item_code}** | **Color Name :{color_name}** |**Material :{material}**	|
| **Po.No.:{po_number}**| 	**Lot size :{lot_size}** | **Q'TY Cavity :{qty_cavity}** | **Drawing No :{drawing_number}**	|

## Contents of inspection

|Item Inspection           | Contents of Inspection        | Standard           | Measuring Method  |No.1|No.2|No.3|No.4|
| -------------- | ----------------- | ---------------- | -------------- |-------------- |-------------- |-------------- |-------------- |
|Appearance         | 	        |          |   |||||
|      | 	เช็คความถูกต้องของใบสั่งกับชิ้นงาน        | ถูกต้อง       | Visual |{M1}|{M2}|||
|      | 	ความเหมือนของสี       | ตามตัวอย่าง       | Visual   |{N1}|{N2}|||
|      | 	 ประกอบร่วม	      | ตามตัวอย่าง       | Visual   |{O1}|{O2}|||
|      | 	 รอยประกบต้องไม่เกิน		 | ไม่มีคราบสนิม       | Visual   |{P1}|{P2}||

## Problem : {problem}



## Measurement Result

| Location | STD      | No.1      | No.2          | No.3 | No.4  |Result|
| ----------------- | ---------------- | -------------- |-------------- |-------------- |--------------|--------------|
| 	A      | {STD_A} |     {A1}    | {A2}  | {A3}  | {A4} |{result_A}|
| 	   B    | {STD_B}|     {B1}    | {B2}  | {B3}  |{B4} |{result_B}|
| 	    C    | {STD_C} |     {C1}    | {C2}  | {C3}  | {C4} |{result_C}|
| 	    D    | {STD_D}       |   {D1}      | {D2}  |{D3}   | {D4} |{result_D}|
| 	    E    |   {STD_E}      |   {E1}      | {E2}  |{E3}   |{E4} |{result_E}|
| 	    F    |   {STD_F}      |   {F1}      | {F2}  |{F3}   |{F4} |{result_F}|
| 	    G    |   {STD_G}      |    {G1}     | {G2}  | {G3}  |{G4}  |{result_G}|
| 	    H    |   {STD_H}      |   {H1}      | {H2}  | {H3}  |{H4}  |{result_H}|
| 	    I   |   {STD_I}      |   {I1}      | {I2}  | {I3}  | {I4} |{result_I}|
| 	    J   |   {STD_J}      |    {J1}     | {J2}  | {J3}  | {J4} |{result_J}|
| 	    K   |   {STD_K}      |    {K1}     | {K2}  | {K3}  |{K4}  |{result_K}|
| 	    L   |   {STD_L}      |    {L1}     | {L2}  | {L3}  | {L4} |{result_L}|

## Summary


|หมายเหตุ : วิธีการตรวจสอบให้อ้างอิงตามคู่มือที่กำหนด การตรวจสอบบันทึกผลที่ถูกต้องเป็น OK  การตรวจสอบบันทึกผลที่ไม่ถูกต้องเป็น NG  |Inspection by: {inspection_by}      | 	 Approve by: |
| --------------------- | -------------- | ----------------- |
| **Remark: {remark}**   |**Pass/Accept/Reject:**      | 	**{status}**    |
'''.format(worker_name = worker_name,machine_name = machine_name,partname_model= partname_model,document_no = document_no,
           inspection_date = inspection_date,item_code = item_code,color_name = color_name,material = material,po_number = po_number,lot_size = lot_size,qty_cavity = qty_cavity,
           drawing_number = drawing_number,M1 = M1 ,M2 = M2, N1 = N1,N2 = N2,O1 =O1,O2 = O2, P1 = P1,P2 = P2,
           problem = problem,STD_A = STD_A,A1 = A1,A2 = A2,A3 = A3,A4 = A4,result_A = result_A,STD_B = STD_B,B1 = B1,B2 = B2,B3 = B3,B4 = B4,result_B = result_B,
           STD_C = STD_C,C1 = C1,C2 = C2,C3 = C3,C4 = C4,result_C = result_C,STD_D = STD_D,D1 = D1,D2 = D2,D3 = D3,D4 = D4,result_D = result_D,STD_E = STD_E,E1 = E1,E2 = E2,
           E3 = E3,E4 = E4,result_E =result_E,STD_F = STD_F,F1 = F1,F2 = F2,F3 = F3,F4 = F4,result_F = result_F,STD_G = STD_G,G1 = G1,G2 = G2,G3 = G3,G4 = G4,result_G = result_G ,
STD_H = STD_H,H1 = H1,H2 = H2,H3 = H3,H4 = H4,result_H = result_H,STD_I = STD_I,I1 = I1,I2 = I2,I3 = I3,I4 = I4,result_I = result_I,
STD_J = STD_J,J1 = J1,J2 = J2,J3 = J3,J4 = J4,result_J = result_J,STD_K = STD_K,K1 = K1,K2 = K2,K3 = K3,K4 = K4,result_K = result_K,STD_L = STD_L,L1 = L1,L2 = L2,L3 = L3,L4 = L4,
result_L = result_L,inspection_by = inspection_by,remark = remark,status = status)

In [0]:
print(output_template)


|WANAVIT MANUFACTURING CO.,LTD  |
|:-------------------------------:|
| **IPQC. PLASTIC DATA SHEET : ( First Trial )**  |

| Worker Name :อีด  | M/C:35     |  Partname/Model :ก้านยึดใบชัดเตอร์ด้านหน้า ac10r1   | Document .No. :2597 |
|----------|-------------|------|------|
| **Inspection date&Time :3/6/2020**|  **Item Code :3410010** | **Color Name :ดำ** |**Material :ABS**	|
| **Po.No.:1m2006-015-2012**| 	**Lot size :7500** | **Q'TY Cavity :2** | **Drawing No :cooler-011**	|

## Contents of inspection

|Item Inspection           | Contents of Inspection        | Standard           | Measuring Method  |No.1|No.2|No.3|No.4|
| -------------- | ----------------- | ---------------- | -------------- |-------------- |-------------- |-------------- |-------------- |
|Appearance         | 	        |          |   |||||
|      | 	เช็คความถูกต้องของใบสั่งกับชิ้นงาน        | ถูกต้อง       | Visual |OK1| OK2|||
|      | 	ความเหมือนของสี       | ตามตัวอย่าง       | Visual   |OK1| OK2|||
|      | 	 


|WANAVIT MANUFACTURING CO.,LTD  |
|:-------------------------------:|
| **IPQC. PLASTIC DATA SHEET : ( First Trial )**  |

| Worker Name :อีด  | M/C:35     |  Partname/Model :ก้านยึดใบชัดเตอร์ด้านหน้า ac10r1   | Document .No. :2597 |
|----------|-------------|------|------|
| **Inspection date&Time :3/6/2020**|  **Item Code :3410010** | **Color Name :ดำ** |**Material :ABS**	|
| **Po.No.:1m2006-015-2012**| 	**Lot size :7500** | **Q'TY Cavity :2** | **Drawing No :cooler-011**	|

## Contents of inspection

|Item Inspection           | Contents of Inspection        | Standard           | Measuring Method  |No.1|No.2|No.3|No.4|
| -------------- | ----------------- | ---------------- | -------------- |-------------- |-------------- |-------------- |-------------- |
|Appearance         | 	        |          |   |||||
|      | 	เช็คความถูกต้องของใบสั่งกับชิ้นงาน        | ถูกต้อง       | Visual |OK1| OK2|||
|      | 	ความเหมือนของสี       | ตามตัวอย่าง       | Visual   |OK1| OK2|||
|      | 	 ประกอบร่วม	      | ตามตัวอย่าง       | Visual   |OK1| OK2|||
|      | 	 รอยประกบต้องไม่เกิน		 | ไม่มีคราบสนิม       | Visual   |OK1| OK2||

## Problem : 



## Measurement Result

| Location | STD      | No.1      | No.2          | No.3 | No.4  |Result|
| ----------------- | ---------------- | -------------- |-------------- |-------------- |--------------|--------------|
| 	A      | 3.60-3.80 |     3.04    | 3.1  |   |  |NG|
| 	   B    | 2.70-3.00|     2.99    | 2.98  |   | |OK|
| 	    C    | 3.80-4.20 |     4.13    | 4.13  |   |  |OK|
| 	    D    |        |         |   |   |  ||
| 	    E    |         |         |   |   | ||
| 	    F    |         |         |   |   | ||
| 	    G    |         |         |   |   |  ||
| 	    H    |         |         |   |   |  ||
| 	    I   |         |         |   |   |  ||
| 	    J   |         |         |   |   |  ||
| 	    K   |         |         |   |   |  ||
| 	    L   |         |         |   |   |  ||

## Summary


|หมายเหตุ : วิธีการตรวจสอบให้อ้างอิงตามคู่มือที่กำหนด การตรวจสอบบันทึกผลที่ถูกต้องเป็น OK  การตรวจสอบบันทึกผลที่ไม่ถูกต้องเป็น NG  |Inspection by: ipQcpd0016      | 	 Approve by: |
| --------------------- | -------------- | ----------------- |
| **Remark: สุรีรัตน์**   |**Pass/Accept/Reject:**      | 	**Accept**    |

In [0]:

#output_template = '''
|WANAVIT MANUFACTURING CO.,LTD  |
|:-------------------------------:|
| **IPQC. PLASTIC DATA SHEET : ( First Trial )**  |

| Worker Name :{worker_name}  | M/C:{machine_name}     |  Partname/Model :{partname_model}   | Document .No. :{document_no} |
|----------|-------------|------|------|
| **Inspection date&Time :{inspection_date}**|  **Item Code :{item_code}** | **Color Name :{color_name}** |**Material :{material}**	|
| **Po.No.:{po_number}**| 	**Lot size :{lot_size}** | **Q'TY Cavity :{qty_cavity}** | **Drawing No :{drawing_number}**	|

## Contents of inspection

|Item Inspection           | Contents of Inspection        | Standard           | Measuring Method  |No.1|No.2|No.3|No.4|
| -------------- | ----------------- | ---------------- | -------------- |-------------- |-------------- |-------------- |-------------- |
|Appearance         | 	        |          |   |||||
|      | 	เช็คความถูกต้องของใบสั่งกับชิ้นงาน        | ถูกต้อง       | Visual |{M1}|{M2}|{M3}|{M4}|
|      | 	ความเหมือนของสี       | ตามตัวอย่าง       | Visual   |{N1}|{N2}|{N3}|{N4}|
|      | 	 ประกอบร่วม	      | ตามตัวอย่าง       | Visual   |{O1}|{O2}|{O3}|{O4}|
|      | 	 รอยประกบต้องไม่เกิน		 | ไม่มีคราบสนิม       | Visual   |{P1}|{P2}|{P3}|{P4}|

## Problem : {problem}



## Measurement Result

| Location | STD      | No.1      | No.2          | No.3 | No.4  |Result|
| ----------------- | ---------------- | -------------- |-------------- |-------------- |--------------|--------------|
| 	A      | {STD_A} |     {A1}    | {A2}  | {A3}  | {A4} |{result_A}|
| 	   B    | {STD_B}|     {B1}    | {B2}  | {B3}  |{B4} |{result_B}|
| 	    C    | {STD_C} |     {C1}    | {C2}  | {C3}  | {C4} |{result_C}|
| 	    D    | {STD_D}       |   {D1}      | {D2}  |{D3}   | {D4} |{result_D}|
| 	    E    |   {STD_E}      |   {E1}      | {E2}  |{E3}   |{E4} |{result_E}|
| 	    F    |   {STD_F}      |   {F1}      | {F2}  |{F3}   |{F4} |{result_F}|
| 	    G    |   {STD_G}      |    {G1}     | {G2}  | {G3}  |{G4}  |{result_G}|
| 	    H    |   {STD_H}      |   {H1}      | {H2}  | {H3}  |{H4}  |{result_H}|
| 	    I   |   {STD_I}      |   {I1}      | {I2}  | {I3}  | {I4} |{result_I}|
| 	    J   |   {STD_J}      |    {J1}     | {J2}  | {J3}  | {J4} |{result_J}|
| 	    K   |   {STD_K}      |    {K1}     | {K2}  | {K3}  |{K4}  |{result_K}|
| 	    L   |   {STD_L}      |    {L1}     | {L2}  | {L3}  | {L4} |{result_L}|

## Summary


|หมายเหตุ : วิธีการตรวจสอบให้อ้างอิงตามคู่มือที่กำหนด การตรวจสอบบันทึกผลที่ถูกต้องเป็น OK  การตรวจสอบบันทึกผลที่ไม่ถูกต้องเป็น NG  |Inspection by: {inspection_by}      | 	 Approve by: |
| --------------------- | -------------- | ----------------- |
| **Remark: {remark}**   |**Pass/Accept/Reject:**      | 	**{status}**    |
'''.format(worker_name = worker_name,machine_name = machine_name,partname_model= partname_model,document_no = document_no,
           inspection_date = inspection_date,item_code = item_code,color_name = color_name,material = material,po_number = po_number,lot_size = lot_size,qty_cavity = qty_cavity,
           drawing_number = drawing_number,M1 = M1 ,M2 = M2,M3 = M3,M4 = M4, N1 = N1,N2 = N2,N3 = N3 ,N4 = N4,O1 =O1,O2 = O2,O3 = O3,O4 =O4, P1 = P1,P2 = P2,P3 = P3,P4 = P4,
           problem = problem,STD_A = STD_A,A1 = A1,A2 = A2,A3 = A3,A4 = A4,result_A = result_A,STD_B = STD_B,B1 = B1,B2 = B2,B3 = B3,B4 = B4,result_B = result_B,
           STD_C = STD_C,C1 = C1,C2 = C2,C3 = C3,C4 = C4,result_C = result_C,STD_D = STD_D,D1 = D1,D2 = D2,D3 = D3,D4 = D4,result_D = result_D,STD_E = STD_E,E1 = E1,E2 = E2,
           E3 = E3,E4 = E4,result_E =result_E,STD_F = STD_F,F1 = F1,F2 = F2,F3 = F3,F4 = F4,result_F = result_F,STD_G = STD_G,G1 = G1,G2 = G2,G3 = G3,G4 = G4,result_G = result_G ,
STD_H = STD_H,H1 = H1,H2 = H2,H3 = H3,H4 = H4,result_H = result_H,STD_I = STD_I,I1 = I1,I2 = I2,I3 = I3,I4 = I4,result_I = result_I,
STD_J = STD_J,J1 = J1,J2 = J2,J3 = J3,J4 = J4,result_J = result_J,STD_K = STD_K,K1 = K1,K2 = K2,K3 = K3,K4 = K4,result_K = result_K,STD_L = STD_L,L1 = L1,L2 = L2,L3 = L3,L4 = L4,
result_L = result_L,inspection_by = inspection_by,remark = remark,status = status)




NameError: ignored